### PR TITLE
Relabel/identify more variables and functions based on current knowledge

### PR DESCRIPTION
**Pull request by MaciekBaron**
Labelled a few variables and identified the purpose of some of the functions.

**Note**: I took the liberty of renaming `gCharacterSelections` to `gCharacterGridSelections`, as this variable indicates the position on the character selection grid rather than the selected character. I have then used `gCharacterSelections` for the actual array of character picks.

### CODE REVIEW

File: asm/bss_800029B0.s [hunk 1]
```diff
@@ -229,7 +229,7 @@ glabel D_8015F8E4
 glabel D_8015F8E8
 .skip 8
 
-glabel D_8015F8F0
+glabel gPlayerPositionLookup
```
Comment (MegaMech) [inline asm/bss_800029B0.s:232]: Is this a lookup table? If so just put LUT.

File: asm/non_matchings/audio/external/func_800C76C0.s [hunk 2]
```diff
@@ -473,8 +473,8 @@ glabel func_800C76C0
 /* 0C89A4 800C7DA4 93C40043 */   lbu   $a0, 0x43($fp)
 /* 0C89A8 800C7DA8 0C0323BE */  jal   func_800C8EF8
 /* 0C89AC 800C7DAC 24040017 */   li    $a0, 23
-/* 0C89B0 800C7DB0 3C19800E */  lui   $t9, %hi(D_800DC5E8) # $t9, 0x800e
-/* 0C89B4 800C7DB4 8F39C5E8 */  lw    $t9, %lo(D_800DC5E8)($t9)
+/* 0C89B0 800C7DB0 3C19800E */  lui   $t9, %hi(gWinningPlayerIndex) # $t9, 0x800e
```
Comment (MegaMech) [inline asm/non_matchings/audio/external/func_800C76C0.s:476]: `gPlayerWinningIndex`?

File: asm/bss_800029B0.s (hunk 1 above)
Comment (MaciekBaron) [reply asm/bss_800029B0.s]: Done

File: asm/non_matchings/audio/external/func_800C76C0.s (hunk 2 above)
Comment (MaciekBaron) [reply asm/non_matchings/audio/external/func_800C76C0.s]: It's the index of the "winning player", but I guess your version makes sense too 👍

**Comment by MegaMech** [comment]
Edit the comment and I'll merge. I did a lot of arguing with myself on this one (Regarding naming `CharacterSelections`). This is a good change and will prevent future confusion.

**Comment by MaciekBaron** [comment]
Which comment sorry?

**Comment by MegaMech** [comment]
Note that these are not the same as the values
 * found in gCharacterSelections as those are more
 * found in gCharacterGridSelections as those are more
 * akin to indexes than IDs

Edit that to
those are ordered by table
~~akin to indexes than ids~~

**Comment by MaciekBaron** [comment]
Thanks, I think I now did it the way you wanted.

**Review by MegaMech — APPROVED** [review]

**Pull request merged**